app/models/email_authentication.rb

```diff
@@ -24,4 +24,11 @@ class EmailAuthentication < ApplicationRecord
   validates :password,
     format: {with: /(?=.*\d)/, error: :needs_number, message: "must contain at least one number"},
     allow_nil: true
+
+  def self.generate_password
```

Comment:
I moved this to more closely associate the logic with the model and make it less likely it would be missed again in the future. 
I'm open to other solutions for how to generate a guaranteed valid password. I didn't want to mess around with retries. I tried this 1,000,000 times (using alphanumeric instead of base64 to improve the odds and adding a random number to eliminate the missing number issue that was the most frequent failure) and still got 4 errors:
`SecureRandom.alphanumeric(24) + SecureRandom.random_number(1).to_s`

Reply:
No, I really like this. Perfect KISS example.

You had also suggested (in Slack I think) that this password doesn't need to be random. Is that true? Is it used so temporarily that it can be a static string, because it's overridden anyway? Or does that make us feel a bit too uncomfortable from a security standpoint?

Reply:
If I said that, I don't think I meant to. But I did just test that theory and it looks like the temporary password is not usable until you accept the invitation by setting a setting a new password. Regardless, I don't think it's worth changing to a static password approach. Who knows what holes might exist in devise.

Reply:
Yeahhh my worry as well .Especially since our codebase is open source.